directives/tabs.py

```diff
@@ -0,0 +1,126 @@
+from sphinx.util import logging
+
+import aplus_nodes
+
+CSS_FILE = 'css/aplus_tab.css'
+JS_FILE = 'js/aplus_tab.js'
+
+assets_path = 'static'
+
+logger = logging.getLogger(__name__)
+
+
+class NavTabDirective(Directive):
+    has_content = True
+    optional_arguments = 1
+
+    def run(self):
+        self.assert_has_content()
+        text = '\n'.join(self.content)
+        node = nodes.container(text)
+        node['classes'].append('aplus-tabs')
+
+        if self.arguments and self.arguments[0]:
+            node['classes'].append(self.arguments[0])
+
+        self.add_name(node)
+        self.state.nested_parse(self.content, self.content_offset, node)
+        return [node]
+
+
+class TabContentDirective(Directive):
+    has_content = True
+    option_spec = {'title': directives.unchanged}
+    required_arguments = 1
+
+    def run(self):
+        self.assert_has_content()
+        text = '\n'.join(self.content)
+        node = nodes.container(text)
+        node['ids'].append('tab-%s' % self.arguments[0])
+        node['classes'].append('tab-content')
+
+        par = nodes.paragraph(text=self.options["title"])
+        par['classes'].append('tab-title')
+        node += par
+
+        self.add_name(node)
+        self.state.nested_parse(self.content, self.content_offset, node)
+
+        return [node]
+
+def add_assets(app):
+    # This method reads the `include_annotated_css` and `include_annotated_js`
+    # settings from the conf.py file located in the course directory. If such
+    # settings are not found, the default settings defined in the setup()
+    #  method will be used instead
+    app.config.include_tab_css and app.add_stylesheet(CSS_FILE)
+    app.config.include_tab_js and app.add_javascript(JS_FILE)
+
+
+def copy_asset_files(app, exception):
+    if exception:
+        return
+
+        
+    # The files are added to the _build/html/_static/css folder.
+    if app.config.include_tab_css:
+        logger.info('Copying CSS files from the tabs directive to the _static folder... ')
+        html_static_path_css = os.path.join(assets_path, CSS_FILE)
+        local_path_css = os.path.join(os.path.dirname(__file__), html_static_path_css)
+        copy_asset(local_path_css, os.path.join(app.outdir, '_static', 'css'))
+        logger.info('done')
+
+    # The files are added to the _build/html/_static/js folder.
+    if app.config.include_tab_js:
+        logger.info(
+        'Copying JS files from the annotated directive to the _static folder... ')
+        html_static_path_js = os.path.join(assets_path, JS_FILE)
+        local_path_js = os.path.join(os.path.dirname(__file__), html_static_path_js)
+        copy_asset(local_path_js, os.path.join(app.outdir, '_static', 'js'))
+        logger.info('done')
+
+
+def setup(app):
+
+    app.add_config_value('include_tab_css', True, 'html')
+    app.add_config_value('include_tab_js', True, 'html')
+
+    app.add_directive('aplus-tabs',  NavTabDirective)
```

Comment:
Why is the directive named `aplus-tabs`? It is not really tied to A+ at all, is it? If this directive could be very well used outside A+, then should the name be more generic without any connection to A+?

Reply:
I will use the `rst-tabs` name instead